File: examples/rest/nodejs/sample_client.js
import request from 'needle';

( async () => {
const color = {
    PURPLE : '\x1b[95m',
    CYAN : '\x1b[96m',
    DARKCYAN : '\x1b[36m',
    BLUE : '\x1b[94m',
    GREEN : '\x1b[92m',
    YELLOW : '\x1b[93m',
    RED : '\x1b[91m',
    BOLD : '\x1b[1m',
    UNDERLINE : '\x1b[4m',
    END : '\x1b[0m'
};

let payload = {
    model : 'vicuna-v1-7b',
    messages: [{"role": "user", "content": "Write a haiku"}],
    stream: false
};

const print = ( str ) => {
    process.stdout.write(str);
};

const newline = () => {
    print('\n');
}

newline();
print(color.BOLD + "Without streaming:" + color.END);
newline();

let r = await request("post", "http://127.0.0.1:8000/v1/chat/completions", payload, {json: true});

print(color.GREEN + r.body.choices[0].message.content + color.END);
print('\n');
// Reset the chat
r = await request("post", "http://127.0.0.1:8000/v1/chat/completions", payload, {json: true});
print(color.BOLD + "Reset chat" + color.END);
newline();

// Get a response using a prompt with streaming

payload = {
    "model": "vicuna-v1-7b",
    "messages": [{"role": "user", "content": "Write a haiku"}],
    "stream": true
}

print( color.BOLD + "With streaming:" + color.END);
newline();
r =  request.post( "http://127.0.0.1:8000/v1/chat/completions", payload, {json: true})
.on('readable', function() {
    let jsData = '';
    let data = '';
    while (data = this.read()) {
       const chunk = data.toString().substring(6);
       if (chunk.trim() === "[DONE]")  break;
       jsData = JSON.parse(chunk);
       print(color.GREEN + jsData.choices[0].delta.content + color.END);
    }
})
.on('done', async function () {
    newline();
    let txtresp = await request("get", "http://127.0.0.1:8000/stats");
    print(color.BOLD + "Runtime stats:" + color.END + txtresp.body);

})

})()

File: examples/rest/python/sample_client.py
import json

import requests


class color:
    PURPLE = "\033[95m"
    CYAN = "\033[96m"
    DARKCYAN = "\033[36m"
    BLUE = "\033[94m"
    GREEN = "\033[92m"
    YELLOW = "\033[93m"
    RED = "\033[91m"
    BOLD = "\033[1m"
    UNDERLINE = "\033[4m"
    END = "\033[0m"


# Get a response using a prompt without streaming
payload = {
    "model": "vicuna-v1-7b",
    "messages": [{"role": "user", "content": "Write a haiku"}],
    "stream": False,
}
r = requests.post("http://127.0.0.1:8000/v1/chat/completions", json=payload)
print(
    f"{color.BOLD}Without streaming:{color.END}\n{color.GREEN}{r.json()['choices'][0]['message']['content']}{color.END}\n"
)

# Reset the chat
r = requests.post("http://127.0.0.1:8000/chat/reset", json=payload)
print(f"{color.BOLD}Reset chat:{color.END} {str(r)}\n")

# Get a response using a prompt with streaming
payload = {
    "model": "vicuna-v1-7b",
    "messages": [{"role": "user", "content": "Write a haiku"}],
    "stream": True,
}
with requests.post("http://127.0.0.1:8000/v1/chat/completions", json=payload, stream=True) as r:
    print(f"{color.BOLD}With streaming:{color.END}")
    for chunk in r:
        if chunk[6:].decode("utf-8").strip() == "[DONE]":
            break
        content = json.loads(chunk[6:])["choices"][0]["delta"].get("content", "")
        print(f"{color.GREEN}{content}{color.END}", end="", flush=True)
    print("\n")

# Get the latest runtime stats
r = requests.get("http://127.0.0.1:8000/stats")
print(f"{color.BOLD}Runtime stats:{color.END} {r.json()}\n")
